src/prefect/engine.py

```diff
@@ -1048,6 +1077,9 @@ async def _in_process_pause(
             f"Flow run cannot be paused. Received non-paused state from API: {state}"
         )
 
+    if wait_for_input:
+        await wait_for_input.save(run_input_keyset)
```

Comment:
This confused me and I'm not not sure why. I haven't reached the definition of this method yet.

Reply:
Is this so overridden defaults for title and stuff are available when the user is filling out the form?

Reply:
Right, this is where the schema is actually saved and stored in the input api. It happens down here so that we're not doing it unless our state change is actually accepted. I'll add some commentary here.